spec/controllers/dashboard_controller_spec.rb

```diff
@@ -59,6 +61,80 @@
     end
   end
 
+  describe "GET /autoyast" do
+    before do
+      allow(Velum::SUSEConnect).to receive(:config).and_return(
+        Velum::SUSEConnect::SUSEConnectConfig.new("https://scc.suse.com", "validregcode")
+      )
+    end
+
+    it "if not logged in serves the autoyast content" do
+      VCR.use_cassette("suse_connect/caasp_registration_active", record: :none) do
+        get :autoyast
+        expect(response.status).to eq 200
+      end
+    end
+
+    context "when logged in" do
+      before do
+        sign_in user
+      end
+
+      it "serves the autoyast content" do
+        VCR.use_cassette("suse_connect/caasp_registration_active", record: :none) do
+          get :autoyast
+          expect(response.status).to eq 200
+        end
+      end
+    end
+
+    context "if the credentials are missing for SMT or SCC" do
+      before do
+        allow(Velum::SUSEConnect).to receive(:config).and_raise(
+          Velum::SUSEConnect::MissingCredentialsException
+        )
+      end
+
+      it "serves the autoyast content" do
+        VCR.use_cassette("suse_connect/caasp_registration_active", record: :none) do
+          get :autoyast
+          expect(response.status).to eq 200
+        end
+      end
+    end
+
+    context "if the registration code is missing" do
+      before do
+        allow(Velum::SUSEConnect).to receive(:config).and_raise(
+          Velum::SUSEConnect::MissingRegCodeException
+        )
+      end
+
+      it "serves the autoyast content" do
+        VCR.use_cassette("suse_connect/caasp_registration_active", record: :none) do
+          get :autoyast
+          expect(response.status).to eq 200
+        end
+      end
+    end
+
+    context "if there is a connectivity problem with SMT or SCC" do
+      before do
+        allow(Velum::SUSEConnect).to receive(:config).and_raise(
+          Velum::SUSEConnect::SCCConnectionException
+        )
+      end
+
+      it "renders a 503 status and a blank page" do
```

Comment:
I'd say so... I have never had access to a real `SMT` instance from where I could take some responses with VCR and play with it...